example/lib/adapter/ble_adapter.dart

```diff
@@ -40,23 +40,17 @@ class BleAdapter {
   }
 
   Stream<BlePeripheral> startPeripheralScan() {
-    return _bleManager.startPeripheralScan().transform(
-      new StreamTransformer<ScanResult, BlePeripheral>.fromHandlers(
-        handleData: (ScanResult scanResult, EventSink<BlePeripheral> sink) {
-          if (scanResult.advertisementData.localName != null) {
-            final peripheral = BlePeripheral(
-              scanResult.peripheral.name,
-              scanResult.peripheral.identifier,
-              scanResult.rssi,
-              false,
-              BlePeripheralCategoryResolver.categoryForName(
-                  scanResult.peripheral.name),
-            );
-            sink.add(peripheral);
-          }
-        },
-      ),
-    );
+    return _bleManager.startPeripheralScan().map((scanResult) {
+      return BlePeripheral(
+          scanResult.peripheral.name ??
+              scanResult.advertisementData.localName ??
+              'Unknown',
+          scanResult.peripheral.identifier,
+          scanResult.rssi,
+          false,
+          BlePeripheralCategoryResolver.categoryForName(
```

Comment:
`.categoryForScanResult(ScanResult)`? You're only using name, but it doesn't really discern between different types and passing whole scan result lets the resolver decide what to use.

Reply:
I agree and I would like to address this in a separate PR, because this one is rather about removing the null-check of `scanResult.advertisementData.localName` (added a description to this PR just today, probably after you've already reviewed it).

Reply:
Created PR for that #89